jplag/src/main/java/de/jplag/clustering/ClusteringFactory.java

```diff
@@ -31,25 +29,11 @@
         }
 
         // init preprocessor
-        ClusteringPreprocessor preprocessor;
-        switch (options.getPreprocessor()) {
-            case CUMULATIVE_DISTRIBUTION_FUNCTION:
-                preprocessor = new CumulativeDistributionFunctionPreprocessor();
-                break;
-            case THRESHOLD:
-                preprocessor = new ThresholdPreprocessor(options.getPreprocessorThreshold());
-                break;
-            case PERCENTILE:
-                preprocessor = new PercentileThresholdProcessor(options.getPreprocessorPercentile());
-                break;
-            case NONE:
-            default:
-                preprocessor = null;
-                break;
-        }
-        if (preprocessor != null) {
+        Optional<ClusteringPreprocessor> preprocessor = options.getPreprocessor().constructPreprocessor(options);
+
+        if (preprocessor.isPresent()) {
             // Package preprocessor into a clustering algorithm
-            clusteringAlgorithm = new PreprocessedClusteringAlgorithm(clusteringAlgorithm, preprocessor);
+            clusteringAlgorithm = new PreprocessedClusteringAlgorithm(clusteringAlgorithm, preprocessor.orElseThrow());
```

Comment:
why orElseThrow? Simply use get()

Reply:
From the Optional#get() API doc:
> The preferred alternative to this method is [orElseThrow()](https://docs.oracle.com/javase/10/docs/api/java/util/Optional.html#orElseThrow()).

Reply:
See also https://bugs.openjdk.java.net/browse/JDK-8140281 on why this is the case.